tests/mocked_plans/test_base_plan.py

```diff
@@ -32,3 +33,27 @@ async def test_base_plan(_, sample_plans):
     plan = await generate_plan(analysis_results, args)
 
     assert str(plan) == exp_plan
+
+
+@pytest.mark.asyncio
+@patch("cou.utils.nova_compute.get_instance_count", side_effect=[1, 0, 1])
+async def test_base_plan_non_empty(_, sample_plans, caplog):
+    """Testing non-empty hypervisor logs."""
+    args = CLIargs("plan", auto_approve=True)
+    model, __ = sample_plans["base.yaml"]
+
+    with caplog.at_level(logging.INFO):
+        analysis_results = await Analysis.create(model)
+        await generate_plan(analysis_results, args)
+
+    expected_skipped_log = "Skipped (non-empty) hypervisors: nova-compute/0, nova-compute/2"
```

Comment:
I think it's better to use something like:

```python
expected_skipped_log = ("'cou.steps.plan'", logging.INFO, "Skipped (non-empty) hypervisors: nova-compute/0, nova-compute/2")
```
and then:

```python
assert expected_skipped_log in caplog.record_tuples
```

This way we guarantee the place that the logger was used, the log level and the message